sycl/include/CL/sycl/detail/pi.h

```diff
@@ -42,6 +42,16 @@
 #define _PI_CONCAT(a, b) _PI_STRING_HELPER(a.b)
 #define _PI_H_VERSION_STRING                                                   \
   _PI_CONCAT(_PI_H_VERSION_MAJOR, _PI_H_VERSION_MINOR)
+
+// These are Intel-specific extensions.
+#define EXT_INTEL_PCI_ADDRESS 0x5000
+#define EXT_INTEL_GPU_EU_COUNT 0x5001
+#define EXT_INTEL_GPU_EU_SIMD_WIDTH 0x5002
+#define EXT_INTEL_GPU_SLICES 0x5003
```

Comment:
These macro should be undef'ed.

Reply:
Done.

Reply:
Why do we need these macros at all? Why not just put the values to the corresponding PI_DEVICE_INFO_... values? BTW, how are we sure values >0x5000 don't overlap with something we already have from OpenCL?

Reply:
I could assign the values directly in the enum.  The way I have done it is the way I normally see it done.  

Visual inspection is the only way to tell if there is an overlap.  This is fragile.  I had thought that cl_ext_intel.h was part of this repository, and these values could be added to it.  I recently learned that cl_ext_intel.h is an external dependency, and has its own process requiring an extension and review process to update.  @jbrodman did not think that Intel-specific extensions would be accepted.

Reply:
I'd prefer no macros unless they serve very good purpose, which these don't. Are we at least sure that 0x5000 is past all the values in the cl_ext_intel.h too? Maybe we should create a pre-define constant where all the PI extensions (from OpenCL) will be after? I'd suggest starting at 0x10000 making it clear from the code that this is an extension (has bits in the upper half of 32-bit)

Reply:
OK.

Reply:
there was a conflict with 0x10004, so I started the sequence later.